test/mocha-fixtures.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const mms = require('mongodb-memory-server');
+
+/*
+ * Default MMS mongodb version is used, unless MONGOMS_VERSION is set (which is set with the matrix in test.yml for CI)
+ */
+
+// a single-instance running for single-instance tests
+let mongoinstance;
+
+// a replset-instance for running repl-set tests
+let mongorreplset;
+
+// decide wheter to start a in-memory or not
+const startMemoryInstance = !process.env.MONGOOSE_TEST_URI;
+const startMemoryReplset = !process.env.MONGOOSE_REPLSET_URI;
+
+module.exports.mochaGlobalSetup = async function mochaGlobalSetup() {
+  let instanceuri;
+  let replseturi;
+
+  process.env.RUNTIME_DOWNLOAD = '1'; // ensure MMS is able to download binaries in this context
+
+  // set some options when running in a CI
+  if (process.env.CI) {
+    process.env.MONGOMS_PREFER_GLOBAL_PATH = '1'; // set MMS to use "~/.cache/mongodb-binaries" even when the path does not yet exist
+  }
+
+  if (startMemoryInstance) { // Config to decided if an mongodb-memory-server instance should be used
+    // it's needed in global space, because we don't want to create a new instance every test-suite
+    mongoinstance = await mms.MongoMemoryServer.create({ instance: { args: ['--setParameter', 'ttlMonitorSleepSecs=1'] } });
+    const uri = mongoinstance.getUri();
+    instanceuri = uri.slice(0, uri.lastIndexOf('/'));
+  } else {
+    instanceuri = process.env.MONGOOSE_TEST_URI;
+  }
+
+  if (startMemoryReplset) {
+    mongorreplset = await mms.MongoMemoryReplSet.create({ replSet: { count: 3, args: ['--setParameter', 'ttlMonitorSleepSecs=1'] } }); // using 3 because even numbers can lead to vote problems
```

Comment:
Can we do `count: 1`? `count: 3` takes longer to start up and we don't do things that would require `count: 3`, like testing replica set failover.

Reply:
i changed it to `3` because the previous value was `2`, and mongodb recommends to not use EVEN numbers, so instead of going down to 1 i had gone up to 2, see https://github.com/Automattic/mongoose/blob/97c3bdb2260addb0d5a25f12bc58ded10aefdbe0/test/common.js#L234 for previous code